index.js

```diff
@@ -0,0 +1,114 @@
+import * as core from "@actions/core";
+import fetch from "node-fetch";
+
+const NETLIFY_BASE_URL = `https://api.netlify.com/api/v1/`;
+const READY_STATES = ["ready", "current"];
+const WAIT_TIMEOUT = 60 * 15; // 15 min
+const WAIT_INCREMENT = 15; // seconds
+
+const netlifyFetch = async (netlifyToken, url) => {
+  return await fetch(NETLIFY_BASE_URL + url, {
+    headers: {
+      Authorization: `Bearer ${netlifyToken}`,
+    },
+  }).then((response) => response.json());
+};
+
+const getDeploy = async (inputs) => {
+  const deploys = await netlifyFetch(
+    inputs.netlifyToken,
+    `sites/${inputs.siteId}/deploys?branch=${inputs.branch}`
+  );
+
+  return deploys.find(
+    (d) => d.context == "production" && d.commit_ref == inputs.commitSha
```

Comment:
minor thing but most eslint rulesets prohibit double equals, since they don't do type checks. `true == 1` = true, `true === 1` = false.
```suggestion
    (d) => d.context === "production" && d.commit_ref === inputs.commitSha
```

Reply:
Done.